cpp_utils/src/cpp/ReturnCode.cpp

```diff
@@ -17,40 +17,57 @@
  *
  */
 
-#include <assert.h>
+#include <cassert>
+#include <map>
+#include <string>
+#include <iostream>
+
 #include <cpp_utils/ReturnCode.hpp>
 
 namespace eprosima {
 namespace utils {
 
-const std::map<ReturnCode, std::string> ReturnCode::to_string_conversion_ =
+ReturnCode::ReturnCode(
+        const fastdds::dds::ReturnCode_t& value)
 {
-    {ReturnCode::RETCODE_OK, "Ok"},
-    {ReturnCode::RETCODE_ERROR, "Error"},
-    {ReturnCode::RETCODE_UNSUPPORTED, "Unsupported"},
-    {ReturnCode::RETCODE_BAD_PARAMETER, "BadParameter"},
-    {ReturnCode::RETCODE_PRECONDITION_NOT_MET, "PreconditionNotMet"},
-    {ReturnCode::RETCODE_OUT_OF_RESOURCES, "OutOfResources"},
-    {ReturnCode::RETCODE_NOT_ENABLED, "NotEnabled"},
-    {ReturnCode::RETCODE_IMMUTABLE_POLICY, "ImmutablePolicy"},
-    {ReturnCode::RETCODE_INCONSISTENT_POLICY, "IncosistentPolicy"},
-    {ReturnCode::RETCODE_ALREADY_DELETED, "AlreadyDeleted"},
-    {ReturnCode::RETCODE_TIMEOUT, "Timeout"},
-    {ReturnCode::RETCODE_NO_DATA, "NoData"},
-    {ReturnCode::RETCODE_ILLEGAL_OPERATION, "IllegalOperation"},
-    {ReturnCode::RETCODE_NOT_ALLOWED_BY_SECURITY, "NotAllowedBySecurity"},
-};
+    switch (value)
+    {
+        case fastdds::dds::RETCODE_OK:
+            value_ = ReturnCode::OK;
+            break;
+        case fastdds::dds::RETCODE_ERROR:
+            value_ = ReturnCode::ERROR;
+            break;
+        case fastdds::dds::RETCODE_NO_DATA:
+            value_ = ReturnCode::NO_DATA;
+            break;
+        case fastdds::dds::RETCODE_NOT_ENABLED:
+            value_ = ReturnCode::NOT_ENABLED;
+            break;
+        case fastdds::dds::RETCODE_PRECONDITION_NOT_MET:
+            value_ = ReturnCode::PRECONDITION_NOT_MET;
+            break;
+        default:
+            value_ = ReturnCode::UNKNOWN;
+            break;
```

Comment:
Even if we do not map all fastdds return code to similar return codes in cpp_utils, I'll add here all Fast DDS codes and map them to the ones interesting for us. So for example, the ones missing should be remapped to `ReturnCode::ERROR` instead of `ReturnCode::UNKNOWN`